### PR TITLE
Cleanup the debounce function

### PR DESCRIPTION
**Pull request by jridgewell**

### CODE REVIEW

File: underscore.js [hunk 1]
```diff
@@ -841,35 +841,25 @@
...
       var callNow = immediate && !timeout;
-      if (!timeout) timeout = setTimeout(later, wait);
+      if (timeout) clearTimeout(timeout);
       if (callNow) {
-        result = func.apply(context, args);
-        context = args = null;
+        timeout = setTimeout(later, wait);
+        result = func.apply(this, args);
+      } else if (!immediate) {
+        timeout = setTimeout(later, wait, this, args);
```
Comment (megawac) [inline underscore.js]: This won't work in pre-es5

Comment (megawac) [inline underscore.js]: This is the reason `_.delay` exists

Comment (jridgewell) [inline underscore.js]: Well that's lame. :disappointed:

Comment (megawac) [inline underscore.js]: You can replace `setTimeout` with delay here and see how it performs. May be worth it, this changeset is pretty nice

**Pull request closed without merging**